### ISSUE TITLE
training code

### ISSUE THREAD

**Issue by wassname** [Jiaxin-Wen/Unsupervised-Elicitation#4]
This is very interesting work! I'm keen to try it out, but it appears there is no training code yet, it's just clustering. 

Could you please upload, or explain the fine-tuning approach? If it's messy (all research code it) you could just put it in a branch.

I'm curious how much longer it took, if you used LoRA, quantization, etc. Did you just train for a single batch for each set?

Also, how did the fine-tuning, vs static version perform? I would expect fine-tuning to be better, but much more complicated and resource intensive to run and implement.

**Comment by wassname** [comment]
Hmm there more code to be found from @imadtyx
- local llama version here https://github.com/imadtyx/Unsupervised-Elicitation
- this is presumably the fine tuning server code https://github.com/imadtyx/axolotl-finetuning/blob/main/Iterative%20Finetuning%20Clean.ipynb, nice!
- but ICM_train.py is missing

**Comment by wassname** [comment]
Oh I misunderstood, it looks like you just train on the new psudo labels!
